### PR TITLE
Github template for release-checklist issue

### PR DESCRIPTION
**Pull request by zippy**
### Summary

Adds github template for release-checklist issue

### TODO:
- [x] CHANGELOGs updated with appropriate info:  N.A.
- [x] All code changes are reflected in docs, including module-level docs: N.A.

### CODE REVIEW

**Review by neonphog — APPROVED** [review]
:+1:

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+name: Release Checklist
+about: Create a release checklist
```
Comment (ThetaSinner) [inline .github/ISSUE_TEMPLATE/release_checklist.md:3]:
```suggestion
about: Release preparation checklist
```

What about this?

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+title: "[RELEASE]"
+labels: ''
+assignees: ''
+
+---
+
+- [ ] confirm works reasonably in the field someplace: @
```
Comment (ThetaSinner) [inline .github/ISSUE_TEMPLATE/release_checklist.md:10]:
```suggestion
- [ ] check for regressions or unexpected behavior by testing with an app that's in active use, in the field: @
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- [ ] confirm launcher build: @
+- [ ] confirm with manual runs of a stress test app: @
+- [ ] confirm scaffolding works: @
+- [ ] confirm holonix branch `main-0.x`: @
+- [ ] confirm any docs/updates ready to publish after release lands: @
```
Comment (ThetaSinner) [inline .github/ISSUE_TEMPLATE/release_checklist.md:15]:
```suggestion
- [ ] migration guide written to help users move to the new release: @
- [ ] confirm any docs/updates ready to publish after release lands: @ 
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+about: Create a release checklist
+title: "[RELEASE]"
+labels: ''
+assignees: ''
+
+---
+
+- [ ] confirm works reasonably in the field someplace: @
+- [ ] confirm launcher build: @
+- [ ] confirm with manual runs of a stress test app: @
```
Comment (jost-s) [inline .github/ISSUE_TEMPLATE/release_checklist.md:12]: We need to define what this going to be, but I suppose it's more of a smoke test than a stress test.
```suggestion
- [ ] confirm with manual runs of a test app: @
```

**Review by jost-s — APPROVED** [review]

**Pull request merged**